apps/server/src/modules/lesson/controller/api-test/lesson-delete.api.spec.ts

```diff
@@ -1,282 +1,274 @@
 		await app.close();
 	});
 
-	describe('Bad requests', () => {
-		describe('when currentUser undefined', () => {
-			it('should return UNAUTHORIZED', async () => {
-				// @ts-expect-error: Test case
-				currentUser = undefined;
-
-				await api.delete(new ObjectId().toHexString()).expect({
-					type: 'UNAUTHORIZED',
-					title: 'Unauthorized',
-					message: 'CurrentUser missing in request context. This route requires jwt authentication guard enabled.',
-					code: 401,
-				});
-			});
-		});
+	describe('delete', () => {
+		describe('when user not exists', () => {
+			const setup = async () => {
+				const lesson = lessonFactory.build();
+				await em.persistAndFlush([lesson]);
+				em.clear();
+
+				return { lessonId: lesson.id };
+			};
 
-		describe('when request param is not mongodb id', () => {
-			let user: User;
+			it('should throw an unauthorized exception', async () => {
+				const { lessonId } = await setup();
 
-			beforeEach(async () => {
-				user = await setupUser([]);
-				currentUser = mapUserToCurrentUser(user);
-			});
+				const response = await request.delete(lessonId);
 
-			it('should return API_VALIDATION_ERROR', async () => {
-				// @ts-expect-error: Test case
-				await api.delete(undefined).expect({
-					type: 'API_VALIDATION_ERROR',
-					title: 'API Validation Error',
-					message: 'API validation failed, see validationErrors for details',
-					code: 400,
-					validationErrors: [{ field: ['lessonId'], errors: ['lessonId must be a mongodb id'] }],
-				});
+				expect(response.statusCode).toEqual(HttpStatus.UNAUTHORIZED);
 			});
 		});
-	});
-
-	describe('Authorization and permissions', () => {
-		describe('when lesson is in course', () => {
-			const setupLessonInCourse = async (
-				teachers: User[] = [],
-				substitutionTeachers: User[] = [],
-				students: User[] = []
-			) => {
-				const course = courseFactory.buildWithId({ teachers, substitutionTeachers, students });
-				const lesson = lessonFactory.buildWithId({ course });
-
-				await em.persistAndFlush([lesson, course]);
-				em.clear();
 
-				return { lesson };
-			};
+		describe('given user is a student', () => {
+			const setupUser = () => UserAndAccountTestFactory.buildStudent();
 
-			describe('when user is teacher', () => {
+			describe('when lesson is part of a course and user is part of it', () => {
 				const setup = async () => {
-					const user = await setupUser([Permission.TOPIC_VIEW]);
-					const { lesson } = await setupLessonInCourse([user]);
-					return lesson;
+					const { studentAccount, studentUser } = setupUser();
+					const course = courseFactory.build({ students: [studentUser] });
+					const lesson = lessonFactory.build({ course });
+
+					await em.persistAndFlush([studentAccount, studentUser, lesson]);
+
+					return { studentAccount, lessonId: lesson.id };
 				};
 
-				it('should return status 200', async () => {
-					const lesson = await setup();
+				it('it should throw a forbidden', async () => {
```

Comment:
Genau genommen testest du hier ja nicht, ob ein Fehler geworfen wird, sondern wie die Response aussieht. Ich finde, das sollte also `it('it should respond with forbidden', ...` oder  `it('it should respond with 403', ...` o.s.ä. heißen.